src/intro/no-std.md

```diff
@@ -1,5 +1,74 @@
 # A `no_std` Rust Environment
 
-> ❌: **This section should cover:**
->
-> * what #[no_std] means: no I/O (filesystems), allocation, etc.
\ No newline at end of file
+As mentioned earlier, this book is all about "bare metal" programming. Standard libraries in turn usually
+require some sort of system (os) integration and higher level functionality, on top of which they implement their functionality.
+Such functionalities are for example I/O (files, networking), Threads, and Heap (dynamic memory allocation).
+
+But don't worry, this does not mean you have to code everything from scratch.
+In an `no_std` environment you still will be able to use the [libcore](https://doc.rust-lang.org/core/),
+if you provide a few basic functions (Either by pulling in in the [rlibc](https://crates.io/crates/rlibc) crate,
+or by implementing them by yourself).
+
+See also:
+* [FAQ](https://www.rust-lang.org/en-US/faq.html#does-rust-work-without-the-standard-library)
+* [RFC-1184](https://github.com/rust-lang/rfcs/blob/master/text/1184-stabilize-no_std.md)
+
+
+## Does this mean I can't use dynamic memory or any kind of I/O?
+No it doesen't! It just means you need to do some work in order to bootstrap it.
+For the cortex-m based setup this book is based on there are even some creates
+which will simplify some of bootstraping.
+
+So if you curios check out this crates:
+* [cortex-m-quickstart](https://crates.io/crates/cortex-m-quickstart)
+* [cortex-m-rt](https://crates.io/crates/cortex-m-rt),
+* [cortex-m-log](https://crates.io/crates/cortex-m-log)
+* [cortex-m-semihosting](https://crates.io/crates/cortex-m-semihosting)
+* [alloc-cortex-m](https://crates.io/crates/alloc-cortex-m)
+* [linked_list_allocator](https://crates.io/crates/linked_list_allocator).
+
+But before you start adding those crates and start allocating consider reading the next section
+`"Static" vs Dynamic memory alloctation` beforehand.
+
+## "Static" vs. Dynmaic memory allocation
+First let us define what we mean with when using the terms "Static" and Dynamic memory allocation in this section.
+
+"Static":
+```
+Means all memory that can be reserved and calculated at compile time,
+assuming no computiation based on dynamic inputs is done (e.g. recursion based on "user input").
```

Comment:
computation

Reply:
... and well, it is it is definitely possible to computation based on dynamic inputs (every input is dynamic, otherwise you'd just be running a completely pre-determined schedule), just the size of the data structures cannot change at runtime.

Reply:
yes you are right, I will rephrase or remove that.